src/main/java/com/snowflake/kafka/connector/internal/IngestSdkProvider.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2023 Snowflake Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.snowflake.kafka.connector.internal;
+
+import com.snowflake.kafka.connector.internal.streaming.StreamingUtils;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClient;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClientFactory;
+import net.snowflake.ingest.utils.SFException;
+import org.apache.kafka.connect.errors.ConnectException;
+
+public class IngestSdkProvider {
+  private static final String STREAMING_CLIENT_PREFIX_NAME = "KC_CLIENT_";
+
+  private LoggerHandler LOGGER;
+  private int streamingIngestClientCount;
+  private SnowflakeStreamingIngestClient streamingIngestClient;
```

Comment:
Instead of having a static IngestSdkProvider, should we have a streamingClient here which could be a static singleton instance? 

My understanding is that we will need exactly one client per node of a connector. This client can be passed on to all tasks in that node.

Reply:
That would be cleaner - it was what i tried at the beginning. However static singletons are impossible to test. Plus if we ever want to dynamically change the number of clients based on load, it would be a much bigger change. 

I think the testability and flexibility of this approach is better than the security of a singleton. Happy to discuss this further if you'd like

Reply:
Ok I have discovered that you can just set a static field during the test. Because of that, I will make the ingestSdkProvider a singleton provider for the client. Thanks for the tip here!

Reply:
Have discovered that you can just set the static singleton during testing - so we will have a static provider singleton. I still think the provider is useful for possible future changes to the client